recipes/stlab/all/conanfile.py

```diff
@@ -0,0 +1,58 @@
+from conans import ConanFile, tools
+from conans.tools import Version
+from conans.errors import ConanInvalidConfiguration
+import os
+
+class Stlab(ConanFile):
+    name = 'stlab'
+    description = 'The Software Technology Lab libraries.'
+    url = 'https://github.com/conan-io/conan-center-index'
+    homepage = 'https://github.com/stlab/libraries'
+    license = 'BSL-1.0'
+    topics = 'conan', 'c++', 'concurrency', 'futures', 'channels'
+
+    settings = 'compiler'
+
+    options = {
+        'coroutines': [True, False]
+    }
+    default_options = {
+        'coroutines': False
+    }
+
+    no_copy_source = True
+    _source_subfolder = 'source_subfolder'
+
+    requires = 'boost/1.69.0'
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "libraries-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, '17')
+
+        if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "9":
+            raise ConanInvalidConfiguration("Need gcc >= 9")
+
+        if self.settings.compiler == "clang" and Version(self.settings.compiler.version) < "8":
+            raise ConanInvalidConfiguration("Need clang >= 8")
+
+        if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) < "19.15":
+            raise ConanInvalidConfiguration("Need MSVC >= 19.15")
```

Comment:
The version we are using for the compiler Visual Studio is not the MSVC one but the Visual Studio one: so VS2019 is 16 and so on. Time ago we started to think about adding a `msvc` compiler to resemble better the actual compiler used, but it is still in the backlog: https://github.com/conan-io/conan/issues/3573

Reply:
You're right, the correct message should be "Need Visual Studio >= 19.15".

Reply:
No, it should be Visual Studio > 16 (it is the one with msvc 19.15, right?)

Reply:
So, the compiler is named visual studio but the version is from msvc?

Reply:
No. The compiler is "Visual Studio" and the version is from the "Visual Studio IDE": 14, 15, 16,...

Reply:
Ok, 15.8 then

Reply:
15.8 then.

Reply:
19.15, 16, 15, 15.8 ... sold!